merlin_standard_lib/utils/misc_utils.py

```diff
@@ -236,4 +236,42 @@ def validate_dataset(paths_or_dataset, batch_size, buffer_size, engine, reader_k
             reader_kwargs["batch_size"] = buffer_size
     else:
         reader_kwargs["part_mem_fraction"] = buffer_size
+
     return Dataset(files, engine=engine, **reader_kwargs)
+
+
+def _augment_schema(
+    schema,
+    cats=None,
+    conts=None,
+    labels=None,
+    sparse_names=None,
+    sparse_max=None,
+    sparse_as_dense=False,
+):
+    from merlin.schema import ColumnSchema, Tags
+
+    labels = [labels] if isinstance(labels, str) else labels
+    for label in labels or []:
+        schema[label] = schema[label].with_tags(Tags.TARGET)
+    for label in cats or []:
+        schema[label] = schema[label].with_tags(Tags.CATEGORICAL)
+    for label in conts or []:
+        schema[label] = schema[label].with_tags(Tags.CONTINUOUS)
+
+    # Set the appropriate properties for the sparse_names/sparse_max/sparse_as_dense
+    for col in sparse_names or []:
+        cs = schema[col]
+        properties = cs.properties
+        if sparse_max and col in sparse_max:
```

Comment:
From what I understand: `sparse_max` is a dict where keys are the sparse feature names and values are the max-length to use for padding, and `sparse_names` is the list of sparse feature names. So is there a use case where the list `sparse_names` is different from the list of the ` sparse_max` keys?

Reply:
This logic was ported over from nvtabular, but I think there are some cases where `sparse_names` and `sparse_max` might be different (or `sparse_max` is under-specified). For example, when we want the value count of a feature to be [unbounded](https://github.com/NVIDIA-Merlin/core/blame/78f1f0b0952fd14b76913b0dd258565c06694abe/merlin/schema/schema.py#L42), we would leave out `sparse_max` but keep `sparse_names`.